php/libraries/Candidate.class.inc

```diff
@@ -42,40 +43,40 @@ class Candidate
      * candID, and will only ever instantiate one object in
      * the event that it's called multiple times.
      *
-     * @param integer $candID A 6 digit Loris CandID that matches
+     * @param CandID $candID A 6 digit Loris CandID that matches
      *                        a candidate in the candidate table
      *
      * @return Candidate the Candidate object
      * @static
      */
-    static public function &singleton(int $candID)
+    static public function &singleton(CandID $candID)
     {
         $candidateList =& $GLOBALS['__candidateObjects'];
-        if (!isset($candidateList[$candID])) {
-            $candidateList[$candID] = new Candidate();
-            $candidateList[$candID]->select($candID);
+        if (!isset($candidateList[strval($candID)])) {
+            $candidateList[strval($candID)] = new Candidate();
+            $candidateList[strval($candID)]->select($candID);
         }
 
-        return $candidateList[$candID];
+        return $candidateList[strval($candID)];
     }
 
     /**
      * This does the work of selecting the candidate and related
      * information from the database and populating it into the
      * Candidate object. Used by singleton method.
      *
-     * @param integer $candID The 6 digit Loris CandID
+     * @param CandID $candID The 6 digit Loris CandID
      *
      * @return void As a side-effect loads candidate into current
      *              object
      * @throws DatabaseException | LorisException
      */
-    public function select(int $candID): void
+    public function select(CandID $candID): void
     {
         $factory = NDB_Factory::singleton();
         $config  = $factory->config();
 
-        $CandArray = array('Candidate' => $candID);
```

Comment:
@driusan is this the best way to get the `value` variable of the CanID object ? we dont want to have a `getValue()` function ?